### DISCUSSION TITLE
Why can't `openxlsx2::wb_get_sheet_names()` do what `openxlsx::getSheetNames()` did?

### DISCUSSION THREAD

**Discussion by SchmidtPaul** [JanMarvin/openxlsx2#775] [Q&A] [answered]
I get what it does now, but this does not make the switch from `openxlsx` to `openxlsx2` easy for me. Is there an alternative?

``` r
example_file <- system.file("extdata", "readTest.xlsx", package = "openxlsx")
openxlsx::getSheetNames(example_file)
#>  [1] "Sheet1"  "Sheet2"  "Sheet 3" "Sheet 4" "Sheet 5" "Sheet 6" "1"      
#>  [8] "11"      "111"     "1111"    "11111"   "111111"
openxlsx2::wb_get_sheet_names(example_file)
#> Error: wb must be class wbWorkbook or R6
```

<sup>Created on 2023-08-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by JanMarvin** [comment, marked as answer]
Hi @SchmidtPaul , the simple solution is "don't switch". We have added many different tweaks all over the place to provide a more consistent interface. Unfortunately there isn't always a one size fits all solution. We might add an internal `wb_load()`, but obviously it's nothing anyone else has had a need for and I'm a bit reluctant to blow up various functions with options not needed and therefore not well tested. But you can create yourself a quick wrapper function.

**Reply by olivroy** [reply]
@JanMarvin to help people make the switch, maybe assert_workbook could hint towards wb_load() ?

Similar to the new behavior of wb_get_named_regions? 
(providing a file to wb_get_named_regions() is now deprecated as openxlsx2 v1.0

**Reply by JanMarvin** [reply]
I'm not sure, maybe I misunderstood the question. I read it as "I don't want to `wb_load()` in the first place" and not as "I don't know what I should do".
Since loading a workbook might take a while in the example file, due to the large sheet size, I was thinking about a special load argument that will read only core, app and workbook, but will skip any worksheets. This could be added as fallback for character arguments here or in #752. In this case additional messages would create only confusion, but maybe I simply got it wrong.

**Comment by SchmidtPaul** [comment]
Alright, thanks. I think I'll simply stick to {readxl} for importing and {openxlsx2} for exporting then.

**Reply by JanMarvin** [reply]
The beauty of R's open source ecosystem is the range of choices. If one package works for task A and another for task B, go for it! If you can use `openxlsx2` wonderful :smile: 

Just for completeness, with `openxlsx2` you get the sheet names like this:

``` r
library(openxlsx2)
example_file <- system.file("extdata", "readTest.xlsx", package = "openxlsx")

wb <- wb_load(example_file)
wb$get_sheet_names()
#>    Sheet1    Sheet2   Sheet 3   Sheet 4   Sheet 5   Sheet 6         1        11 
#>  "Sheet1"  "Sheet2" "Sheet 3" "Sheet 4" "Sheet 5" "Sheet 6"       "1"      "11" 
#>       111      1111     11111    111111 
#>     "111"    "1111"   "11111"  "111111"

# exploit the package :)
wb_load(example_file, sheet = 0)
#> Error in wb_load(example_file, sheet = 0): No such sheet in the workbook. Workbook contains:
#> Sheet1
#> Sheet2
#> Sheet 3
#> Sheet 4
#> Sheet 5
#> Sheet 6
#> 1
#> 11
#> 111
#> 1111
#> 11111
#> 111111
```
